homeassistant/components/lcn/helpers.py

```diff
@@ -336,8 +336,9 @@ async def async_update_config_entry(
     hass: HomeAssistant, config_entry: ConfigEntry
 ) -> None:
     """Fill missing values in config_entry with infos from LCN bus."""
+    new_data = config_entry.data.copy()
     coros = []
-    for device_config in config_entry.data[CONF_DEVICES]:
+    for device_config in new_data[CONF_DEVICES]:
```

Comment:
The copy is only making a copy of the top level dictionary. You're then changing `new_data[CONF_DEVICES]` which is not a copy. You should make copies of `device_config`

Reply:
I decided to make a deepcopy of  `config_entry.data[CONF_DEVICES]`. Shouldn't be much more expensive but keeps the code changes low.